### PR TITLE
Make GetNodeBootstrappingPayload return base64 encoded string.

### PR DESCRIPTION
**Pull request by yizhang4321**
Updated GetAgentKubernetesLabels and GetAgentKubernetesLabelsDeprecated in the function map to generate the resource group names themselves so they don't need to take it from the nodecustomdata.yml. This simplifies the logic for nodecustomdata.yml: we no longer need to "concat" substrings, which contains a variables('labelResourceGroup') in the middle. We can now directly take nodecustomdata.yaml as a string in our code.

### CODE REVIEW

File: pkg/agent/baker.go [hunk 1]
```diff
@@ -26,17 +26,23 @@ func InitializeTemplateGenerator() *TemplateGenerator {
 	return t
 }
 
+// GetNodeBootstrappingPayloadNonARM get node bootstrapping data that's not to be used in an ARM template.
+func (t *TemplateGenerator) GetNodeBootstrappingPayloadNonARM(config *NodeBootstrappingConfiguration) string {
```
Comment (xuto2) [inline pkg/agent/baker.go:30]: I don't think we need to distinguish between the ARM and no ARM implementation, the VM's custom data string should always be base64 string, no matter how we create it

File: pkg/agent/baker.go [hunk 2]
```diff
@@ -48,12 +54,22 @@ func (t *TemplateGenerator) getLinuxNodeCustomDataJSONObject(config *NodeBootstr
...
+	return fmt.Sprintf("{\"customData\": \"[base64('%s')]\"}", t.getLinuxNodeCustomDataJSONObjectCommon(config))
+}
+
+func (t *TemplateGenerator) getLinuxNodeCustomDataJSONObjectNonARM(config *NodeBootstrappingConfiguration) string {
+	customData := t.getLinuxNodeCustomDataJSONObjectCommon(config)
+	base64CustomData := base64.StdEncoding.EncodeToString([]byte(customData))
+	return fmt.Sprintf("{\"customData\": \"%s\"}", base64CustomData)
+}
+
+func (t *TemplateGenerator) getWindowsNodeCustomDataJSONObjectCommon(config *NodeBootstrappingConfiguration) string {
```
Comment (xuto2) [inline pkg/agent/baker.go:72]: not sure if we need to touch windows in this PR - windows cse/custom data is not yet migrated to AgentBaker, what we have here is basically garbage data.

File: pkg/agent/baker.go [hunk 3]
```diff
@@ -192,6 +218,33 @@ func (t *TemplateGenerator) getBakerFuncMap(config *NodeBootstrappingConfigurati
...
+	truncated = strings.ReplaceAll(truncated, ")", "-")
+	const maxLen = 63
+	if len(truncated) > maxLen {
+		truncated = truncated[0:maxLen]
+	}
+
+	if strings.HasSuffix(truncated, "-") ||
+		strings.HasSuffix(truncated, "_") ||
+		strings.HasSuffix(truncated, ".") {
+		return truncated[0:len(truncated)-1] + "z"
```
Comment (xuto2) [inline pkg/agent/baker.go:243]: more precisely:

if len(truncated) > 62 {
 return truncated[0:len(truncated)-1] + "z"
} else {
return truncated + "z"
}
}

Comment (yizhang4321) [reply pkg/agent/baker.go:243]: This is to match the ARM template that sets labelResourceGroup to:
[if(or(or(endsWith(variables('truncatedResourceGroup'), '-'), endsWith(variables('truncatedResourceGroup'), '_')), endsWith(variables('truncatedResourceGroup'), '.')), concat(take(variables('truncatedResourceGroup'), 62), 'z'), variables('truncatedResourceGroup'))]

What it does is: it only replaces the last character with z if it is '-', '_', or '.'.

If truncatedResourceGroup does not end with any of those 3, then
    labelResourceGroup = truncatedResourceGroup

File: pkg/agent/baker.go (hunk 2 above)
Comment (yizhang4321) [reply pkg/agent/baker.go]: I have removed Windows related change.

File: pkg/agent/baker.go (hunk 1 above)
Comment (yizhang4321) [reply pkg/agent/baker.go]: Fixed now. Now we just have a single implementation that gets a base64 encoded bootstrapping custom data.

File: pkg/agent/baker.go [hunk 4]
```diff
@@ -48,11 +51,11 @@ func (t *TemplateGenerator) getLinuxNodeCustomDataJSONObject(config *datamodel.N
 		panic(e)
 	}
 
-	return fmt.Sprintf("{\"customData\": \"[base64(concat('%s'))]\"}", str)
+	return fmt.Sprintf("{\"customData\": \"%s\"}", str)
```
Comment (xuto2) [inline pkg/agent/baker.go:54]: alternatively, we can do base64.StdEncoding.EncodeToString([]byte(str)) here and save the change from caller side.

File: pkg/agent/baker.go (hunk 3 above)
Comment (xuto2) [reply pkg/agent/baker.go:219]: you're missing the conversion of take(variables('truncatedResourceGroup'), 62), where it get max of 62 char  so no trimming for short strings

e.g if the string is "abc-" and the final string will be "abc-z" ,

Comment (yizhang4321) [reply pkg/agent/baker.go]: Fixed now.

**Pull request merged**